base/server/python/pki/server/deployment/__init__.py

```diff
@@ -850,10 +850,13 @@ def import_cert_request(self, subsystem, tag, request):
         request_id_generator = subsystem.config.get('dbs.request.id.generator', 'legacy')
 
         if request_id_generator == 'legacy':
-            # get a new request ID from the server
+            # call the server to generate legacy request ID
             logger.info('Creating request ID for %s cert', tag)
             request.systemCert.requestID = self.client.createRequestID(request)
             logger.info('- request ID: %s', request.systemCert.requestID)
+        else:
```

Comment:
Do we need `elif request_id_generator == 'random'`?

Reply:
I think we have to distinguish between

- legacy -> managed by the server
- all the other -> managed without the server

Even for the moment "all the other" contains only the `random` case I would not limit here so you can implement others without change the python code

Reply:
Yeah, I don't think in the future we will create a new ID generator that can only run inside the server again, so this should cover all future cases (I updated the comment).